client/src/components/assignment_create.js

```diff
@@ -0,0 +1,184 @@
+                    .then((data) => {
+                        let existingAssignments = data.rows
+                        existingAssignments.forEach((existing_a, i) => {
+                            // check for unique assignment name
+                            if (existing_a.name === assignmentName) {
+                                alert(`Assignment with name "${assignmentName}" already exists for this course.`)
+                                throw ""
+                            }
+                        })
+        
+                        // check that deadline is in the future
+                        let deadline_timestamp = new Date(deadline)
+                        const today = new Date();
+                        if (deadline_timestamp <= today.getTime()) {
+                            alert("Assignment deadline must be set to a future time.")
+                            throw ""
+                        }
+                        deadline_timestamp = deadline_timestamp.toISOString()
+        
+                        // compute total max grade from question max grades
+                        var max_grade = 0
+                        questions.forEach((q, i) => {
+                            max_grade += q.max_grade
+                        })
+        
+                        // optional, can be blank
+                        const description = document.getElementById("description").value
+        
+                        // create assignment
+                        fetch(`http://localhost:8080/createAssignment?aid=${next_aid}&cid=${cid}&a_name=${assignmentName}&deadline=${deadline_timestamp}&max_grade=${max_grade}&description=${description}`)
+                        .then((response) => response.json())
+                        .then((data) => {
+                            // get users in course to create assignmentsubmissions for
+                            fetch(`http://localhost:8080/usersInCourse?cid=${cid}`)
+                            .then((response) => response.json())
+                            .then((data) => {
+                                let usersInCourse = data.rows
+                                // create assignmentsubmissions
+                                Promise.all(usersInCourse.map((u) => {
+                                    fetch(`http://localhost:8080/createAssignmentSubmission?uid=${u.student_id}&aid=${next_aid}`)
+                                }))
+                                .then(() => {
+                                    // create questions
+                                    Promise.all(questions.map((q, i) => {
+                                        fetch(`http://localhost:8080/createQuestion?aid=${next_aid}&num=${i+1}&max_grade=${q.max_grade}&description=${q.description}`)
+                                    }))
+                                    .then(() => {
+                                        // create questionsubmissions
```

Comment:
because of how i implemented the student view, you don't have to create questionsubmissions here (because if they exist, the related file is grabbed then displayed) so i think you can remove this particular bit

Reply:
i.e. questionsubmissions are created upon the related assignmentsubmission having is_submitted true, so on submission and not question creation

Reply:
icic that's p good acc since the questionsubmissions were being weird :'v